### PR TITLE
Solved the overlapping of payment options when clicking on 'Add Payment Method'

### PR DESCRIPTION
**Pull request by liyamahendra**
### Details
The payment options were overlapping when they are shown after clicking on the 'Add Payment Method' button.

### Fixed Issues
$ https://github.com/Expensify/App/issues/9283

### Tests
1. Click on the 'Add Payment Method' button2. 
2. Verify that the payment method options pop-over menu is shown below the 'Add Payment Method' button3. 
3. Also verify that the payment method options pop-over menu resizes / displaces appropriately when the browser is resized.

- [x] Verify that no errors appear in the JS console

#### Contributor (PR Author) Checklist
- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there’s a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained “why” the code was doing something instead of only explaining “what” the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named “index.js”. All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] Any functional components have the `displayName` property
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose and it is
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn’t already exist
    - [x] The style can’t be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.

<details>
<summary><h4>PR Reviewer Checklist</h4></summary>

- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] I verified there are no console errors (if there’s a console error not related to the PR, report it or open an issue for it to be fixed)
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained “why” the code was doing something instead of only explaining “what” the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named “index.js”. All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] Any functional components have the `displayName` property
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn’t already exist
    - [ ] The style can’t be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.

</details>

### QA Steps
1. Click on the 'Add Payment Method' button2. 
2. Verify that the payment method options pop-over menu is shown below the 'Add Payment Method' button3. 
3. Also verify that the payment method options pop-over menu resizes / displaces appropriately when the browser is resized.

- [x] Verify that no errors appear in the JS console

#### Web
image

#### Mobile Web
Simulator Screen Shot - iPhone 13 - 2022-06-24 at 01 09 36

#### Desktop
image

#### iOS
Simulator Screen Shot - iPhone 13 - 2022-06-24 at 01 08 38

#### Android

Screenshot_1657641201

### CODE REVIEW

**Comment by marcaaron** [comment]
Can you please update this PR so that it has a more recognizable title? Thanks.

**Comment by liyamahendra** [comment]
@marcaaron Update the title of the PR.

File: src/libs/getClickedTargetLocation/index.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+/**
+ * Returns the Bounding Rectangle for the passed native event's target.
+ *
+ * @param {Object} target
+ * @returns {Object}
+ */
+function getClickedTargetLocation(target) {
```
Comment (mananjadhav) [inline src/libs/getClickedTargetLocation/index.js:7]: @liyamahendra I thought we'll fix the existing function instead of adding a new one also fix this on the KYCWall page?

File: src/pages/settings/Payments/PaymentsPage/BasePaymentsPage.js [hunk 2]
```diff
@@ -86,7 +86,8 @@ class BasePaymentsPage extends React.Component {
         if (!this.state.addPaymentMethodButton) {
             return;
         }
-        const buttonPosition = getClickedElementLocation(this.state.addPaymentMethodButton);
+
```
Comment (mananjadhav) [inline src/pages/settings/Payments/PaymentsPage/BasePaymentsPage.js:89]:
```suggestion
```

**Review by mananjadhav — CHANGES_REQUESTED** [review]
Just one question on retaining the existing function and also fixing the issue for KYCWall.

**Comment by liyamahendra** [comment]
@mananjadhav I did left a [comment](https://github.com/Expensify/App/issues/9283#issuecomment-1156100956) requesting details of `BaseKYCWall` and I can see if we repurpose `getClickedElementLocation`. Would you share some insights please?

> 3. Would you please share insights into how can I test `BaseKYCWall`. Then, I'll make sure I update `getClickedElementLocation` as required.

**Comment by mananjadhav** [comment]
@liyamahendra I'll also have to check how that flow works.

https://github.com/Expensify/App/blob/2595343f78d6d9866684b96f8ba96862b9d2165f/src/pages/settings/Payments/PaymentsPage/BasePaymentsPage.js#L262-L273

A quick search leads me to commenting

**Comment by mananjadhav** [comment]
@liyamahendra Any updates for me on the KYCWall page? Do you still need help with the same?

**Comment by liyamahendra** [comment]
Ah, apologies @mananjadhav , I assumed you'll check the flow works for `KYCWall` and share the details. 

Allow me to review this from my end and see if I can figure this out. If not, I'll ask specific questions please.

**Comment by liyamahendra** [comment]
@mananjadhav I checked the code. Would you mind sharing details as to how can I test / see the `KYCWall` screen rendered in the web app? This will allow me to validate and do the change.

File: src/libs/getClickedTargetLocation/index.js (hunk 1 above)
Comment (liyamahendra) [reply src/libs/getClickedTargetLocation/index.js:7]: `getClickedElementLocation` is only used at two places, so it should be safe to refactor this function. We have a conversation going on regarding `KYCWall`, will do the needful accordingly.

Quick question: I'm towards the opinion that`getClickedTargetLocation` is a more appropriate name rather than `getClickedElementLocation`, what do you think? @mananjadhav

**Comment by mananjadhav** [comment]
> as to how can I test / see the KYCWall screen rendered in the web app?

I don't know the flow either. I can spend some time tomorrow and get you the flow. In my opinion to just verify if the problem exists, we could remove those conditions and force view the page may be by visiting the route?

File: src/libs/getClickedTargetLocation/index.js (hunk 1 above)
Comment (mananjadhav) [reply src/libs/getClickedTargetLocation/index.js:7]: > `getClickedTargetLocation` is a more appropriate name

I agree. Let's discard the old one.

**Comment by mananjadhav** [comment]
I'll check this by tomorrow and share the Test steps with you.

**Comment by mananjadhav** [comment]
@liyamahendra The steps are almost through the force code changes that I asked for (I don't have the betas access)

1. In `BasePaymentPage` force true the conditions the following conditions on L262 and L267.
```
{(true || Permissions.canUseWallet(this.props.betas)) && (
```
and
```
{(true || this.props.userWallet.currentBalance > 0) && (
```
3. Go to the Profile Picture -> RHN (right side bar) -> Payments
4. You'll see a `Transfer Balance` button
5. Hover over it, to see the Payment options.


These steps should help you for the `KYCWall.js`.

**Comment by liyamahendra** [comment]
@mananjadhav thank you for the steps. However, even after doing the changes you suggested, I don't see the `Transfer Balance` button. See the attached screencast.

https://user-images.githubusercontent.com/911879/177257008-d2ab5475-dc04-4d42-baf9-303777c99bfa.mp4

**Comment by liyamahendra** [comment]
@mananjadhav I removed the conditions altogether and was able to see the Transfer Balance Button. While testing it, I don't see the issue with it. See the screencast below. 

https://user-images.githubusercontent.com/911879/177257779-9cb0b1f9-274e-489a-b37c-55437fdd01cc.mp4

Can you advise please?

**Comment by mananjadhav** [comment]
@liyamahendra Goal is to ensure that when we're replacing the function (because it works on the similar logic), lets ensure that we use your function for this component as well.

**Comment by liyamahendra** [comment]
Got it @mananjadhav - since we agreed to remove the former function, I'll test out how this behaves with my function and comment back

**Comment by liyamahendra** [comment]
@mananjadhav have done the changes to `KYCWall`.

**Comment by mananjadhav** [comment]
@liyamahendra I reviewed and tested your changes, but we need to sync the `main` branch and I can take a look at it again. 


Also while the changes are not different parts of the page, I feel we might get some conflicts if https://github.com/Expensify/App/pull/9622 gets merged. I would also personally recommend testing the changes after pulling the latest changes.

Lastly,  please complete the PR checklist. Please mark the items off the checklist even if they don't apply. Tests are mandatory for all Platforms (I can see Android/Native not marked and missing screenshot).

**Comment by liyamahendra** [comment]
> Lastly, please complete the PR checklist. Please mark the items off the checklist even if they don't apply. Tests are mandatory for all Platforms (I can see Android/Native not marked and missing screenshot).

@mananjadhav I uploaded a screenshot for Android and completed the checklist. I also pulled from Main and tested this particular issue. Didn't encounter any conflicts.

**Review by mananjadhav — DISMISSED** [review]
@marcaaron Review done

🎀 👀 🎀
C+ reviewed

<details>
<summary><h4>PR Reviewer Checklist</h4>

The Contributor+ will copy/paste it into a new comment and complete it after the author checklist is completed
</summary>

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/docs/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/docs/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/docs/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/docs/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] Any functional components have the `displayName` property
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

</details>

File: src/libs/getClickedTargetLocation/index.js (hunk 1 above)
Comment (marcaaron) [reply src/libs/getClickedTargetLocation/index.js:7]: I am OK with where this landed, but I think @mananjadhav's first instinct was correct here. Better to focus on the bug fix first and refactoring later.

File: src/pages/settings/Payments/PaymentsPage/BasePaymentsPage.js [hunk 3]
```diff
@@ -125,10 +125,12 @@ class BasePaymentsPage extends React.Component {
      * @param {Boolean} isDefault
      */
     paymentMethodPressed(nativeEvent, accountType, account, isDefault) {
-        const position = getClickedElementLocation(nativeEvent);
+        const position = getClickedTargetLocation(nativeEvent.currentTarget);
+
         this.setState({
-            addPaymentMethodButton: nativeEvent,
+            addPaymentMethodButton: nativeEvent.currentTarget,
```
Comment (marcaaron) [inline src/pages/settings/Payments/PaymentsPage/BasePaymentsPage.js:131]: I know this was not added in this PR, but why are we storing a reference to an HTML element in React component state? This seems like a strange practice.

File: src/pages/settings/Payments/PaymentsPage/BasePaymentsPage.js [hunk 4]
```diff
@@ -203,7 +205,10 @@ class BasePaymentsPage extends React.Component {
      * Hide the add payment modal
      */
     hideAddPaymentMenu() {
-        this.setState({shouldShowAddPaymentMenu: false});
+        this.setState({
+            addPaymentMethodButton: null,
```
Comment (marcaaron) [inline src/pages/settings/Payments/PaymentsPage/BasePaymentsPage.js:209]: What is the purpose of setting this to `null`?

**Review by marcaaron — CHANGES_REQUESTED** [review]
Looks good, but I have a couple of questions that should be answered before we merge.

File: src/pages/settings/Payments/PaymentsPage/BasePaymentsPage.js (hunk 3 above)
Comment (liyamahendra) [reply src/pages/settings/Payments/PaymentsPage/BasePaymentsPage.js:131]: I think we can move this HTML element to a class property. Thoughts? @mananjadhav @marcaaron

Comment (liyamahendra) [reply src/pages/settings/Payments/PaymentsPage/BasePaymentsPage.js:131]: > I am OK with where this landed, but I think @mananjadhav's first instinct was correct here. Better to focus on the bug fix first and refactoring later.

@marcaaron may I know what are you trying to propose please?

Comment (marcaaron) [reply src/pages/settings/Payments/PaymentsPage/BasePaymentsPage.js:131]: That seems better to me as long as we are certain that we are not dependent on the call to `setState()` (e.g. to trigger a re-render for some reason).

File: src/pages/settings/Payments/PaymentsPage/BasePaymentsPage.js (hunk 4 above)
Comment (marcaaron) [reply src/pages/settings/Payments/PaymentsPage/BasePaymentsPage.js:209]: Understood. Not quite an optimization because calling `setState()` triggers a new call to `render()`. If we want to optimize then we should avoid unnecessary calls to `setState()`.

File: src/pages/settings/Payments/PaymentsPage/BasePaymentsPage.js (hunk 3 above)
Comment (mananjadhav) [reply src/pages/settings/Payments/PaymentsPage/BasePaymentsPage.js:131]: I think moving to the class property won't work when we resize the window.

Comment (mananjadhav) [reply src/pages/settings/Payments/PaymentsPage/BasePaymentsPage.js:131]: > > I am OK with where this landed, but I think @mananjadhav's first instinct was correct here. Better to focus on the bug fix first and refactoring later.
> > 
> @marcaaron may I know what are you trying to propose please?

There's not action for this statement. What @marcaaron meant is before we refactor any piece of code (like we renamed the function), we should first focus on fixing the bug. Once we're confident all is working as expected, we can then refactor/enhance the readability of the code. Hope it makes sense :)

Comment (marcaaron) [reply src/pages/settings/Payments/PaymentsPage/BasePaymentsPage.js:131]: >I think moving to the class property won't work when we resize the window. Because we store the element the bounding rectangle gives us the correct coordinates.

Hmm I may have to take a closer look here but my thinking is that when the window is resized that we would have to remeasure the height, but unsure why having the element reference in state makes any difference.

Comment (marcaaron) [reply src/pages/settings/Payments/PaymentsPage/BasePaymentsPage.js:131]: There's some justification [here](https://github.com/Expensify/App/pull/7847#discussion_r816907027) - but it really doesn't make sense to me 🤷‍♂️ 

Anyway, wasn't introduced in this PR so I am fine if we want to do nothing and create a new issue to improve the way this works.

File: src/pages/settings/Payments/PaymentsPage/BasePaymentsPage.js (hunk 4 above)
Comment (marcaaron) [reply src/pages/settings/Payments/PaymentsPage/BasePaymentsPage.js:209]: We can remove it.

Comment (liyamahendra) [reply src/pages/settings/Payments/PaymentsPage/BasePaymentsPage.js]: I've removed this statement.

File: src/pages/settings/Payments/PaymentsPage/BasePaymentsPage.js [hunk 5]
```diff
@@ -203,7 +205,9 @@ class BasePaymentsPage extends React.Component {
      * Hide the add payment modal
      */
     hideAddPaymentMenu() {
-        this.setState({shouldShowAddPaymentMenu: false});
+        this.setState({
+            shouldShowAddPaymentMenu: false,
+        });
```
Comment (marcaaron) [inline src/pages/settings/Payments/PaymentsPage/BasePaymentsPage.js:210]:
```suggestion
        this.setState({shouldShowAddPaymentMenu: false});
```

Unnecessary change

File: src/pages/settings/Payments/PaymentsPage/BasePaymentsPage.js [hunk 6]
```diff
@@ -125,10 +125,12 @@ class BasePaymentsPage extends React.Component {
...
      */
     paymentMethodPressed(nativeEvent, accountType, account, isDefault) {
-        const position = getClickedElementLocation(nativeEvent);
+        const position = getClickedTargetLocation(nativeEvent.currentTarget);
+
         this.setState({
-            addPaymentMethodButton: nativeEvent,
+            addPaymentMethodButton: nativeEvent.currentTarget,
         });
+
```
Comment (marcaaron) [inline src/pages/settings/Payments/PaymentsPage/BasePaymentsPage.js:133]:
```suggestion
```

File: src/pages/settings/Payments/PaymentsPage/BasePaymentsPage.js [hunk 7]
```diff
@@ -125,10 +125,12 @@ class BasePaymentsPage extends React.Component {
      * @param {Boolean} isDefault
      */
     paymentMethodPressed(nativeEvent, accountType, account, isDefault) {
-        const position = getClickedElementLocation(nativeEvent);
+        const position = getClickedTargetLocation(nativeEvent.currentTarget);
+
```
Comment (marcaaron) [inline src/pages/settings/Payments/PaymentsPage/BasePaymentsPage.js:129]:
```suggestion
```

**Review by marcaaron — CHANGES_REQUESTED** [review]

**Comment by mananjadhav** [comment]
I am left with testing it once again. Will do that by today-tomorrow

**Comment by mananjadhav** [comment]
@marcaaron Review done.

🎀 👀 🎀 
C+ reviewed

<details>
<summary><h4>PR Reviewer Checklist</h4>

The Contributor+ will copy/paste it into a new comment and complete it after the author checklist is completed
</summary>

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/docs/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/docs/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/docs/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/docs/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] Any functional components have the `displayName` property
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

</details>

**Review by marcaaron — APPROVED** [review]
